### PR TITLE
[MRG] fix bugs of gw_entropic and armijo to run on gpu

### PR DESCRIPTION
**Pull request by cedricvincentcuaz**
## Types of changes

- fix bug in `ot.gromov._brengman.entropic_gromov_wasserstein` to run on gpu
- fix bug in `ot.optim.line_search_armijo` to run on gpu: the function now converts its inputs into numpy arrays to avoid bugs within scipy.optimize.linesearch.scalar_search_armijo that occurred with cuda variables.

## Motivation and context / Related issue

- complete tests on gpu for the new gromov directory
- correct bug in `line_search_armijo` to run on gpu (cf Issue #445 )

## How has this been tested (if it applies)

- check existing tests for ot.gromov run on gpu + add test in `test_gromov2_gradients` using `line_search_armijo`.
- `test_line_search_armijo_dtype_device` : tests all backends/types for the  `line_search_armijo` function.

## PR checklist

- [x] I have read the [**CONTRIBUTING**](CONTRIBUTING.md) document.
- [x] The documentation is up-to-date with the changes I made (check build artifacts).
- [x] All tests passed, and additional code has been **covered with new tests**.
- [x] I have added the PR and Issue fix to the [**RELEASES.md**](RELEASES.md) file.

### CODE REVIEW

File: ot/optim.py [hunk 1]
```diff
@@ -150,7 +170,10 @@ def generic_conditional_gradient(a, b, M, f, df, reg1, reg2, lp_solver, line_sea
     b : array-like, shape (nt,)
         samples weights in the target domain
     M : array-like, shape (ns, nt)
-        loss matrix
+        loss matrixif isinstance(gfk, int) or isinstance(gfk, float):
```
Comment (rflamary) [inline ot/optim.py:173]: what is this doing in the docstring?

File: ot/optim.py [hunk 2]
```diff
@@ -162,7 +185,7 @@ def generic_conditional_gradient(a, b, M, f, df, reg1, reg2, lp_solver, line_sea
     lp_solver: function,
         linear program solver for direction finding of the (generalized) conditional gradient.
         If set to emd will solve the general regularized OT problem using cg.
-        If set to lp_semi_relaxed_OT will solve the general regularized semi-relaxed OT problem using cg.
+        If set to lp_semi_relaxed_OT will solve the general optimizeregularized semi-relaxed OT problem using cg.
```
Comment (rflamary) [inline ot/optim.py:188]: correct optimizeregularized

**Review by rflamary — COMMENTED** [review]
Just a few changes and then we can merge

File: ot/optim.py (hunk 1 above)
Comment (cedricvincentcuaz) [reply ot/optim.py]: wth... Thanks Rémi for the review ! Should be ready for merge now :)

**Pull request merged**